### ISSUE TITLE
Unable to disable locale routing

### ISSUE THREAD

**Issue by lalitindoria** [vercel/next.js#29000]
### What version of Next.js are you using?

11.1.2

### What version of Node.js are you using?

15.14.0

### What browser are you using?

Chrome

### What operating system are you using?

Ubuntu

### How are you deploying your application?

local using next dev

### Describe the Bug

I do not want to use locale based routing in my next app. Here's my next.config.js

```
/** @type {import('next').NextConfig} */
module.exports = {
  reactStrictMode: true,
  i18n: {
    locales: ["en"],
    defaultLocale: "en",
    localeDetection: false,
  },
  async redirects() {
    return [
      {
        source: "/",
        destination: "/test",
        permanent: false,
        locale: false,
      },
    ];
  },
};
```

Although `localeDetection` has been set to `false` the page is getting redirected from `/` to `/en/test`. Moreover, after adding `locale: false` to my redirect, the redirection is no longer happening.

### Expected Behavior

Expecting redirects to work fine without sub path routing.

### To Reproduce

Can visit this repo - https://github.com/lalitindoria/next-router-locale-bug

**Comment by lalitindoria** [comment]
Also deployed using vercel - https://next-router-locale-bug.vercel.app/

**Comment by Jishnu-Prakash-K** [comment]
Hi, if you don't want  localeDetection, try removing the i18n config altogether from the file. When I tried this, its working as per your requirement. Following is the config that I used. 
`module.exports = {
  reactStrictMode: true,
  async redirects() {
    return [
      {
        source: "/",
        destination: "/test",
        permanent: false,
      },
    ];
  },
};
`
PS: Correct me if I'm wrong :)

**Comment by lalitindoria** [comment]
@Jishnu-Prakash-K Yes that works but I need to be able to disable locale detection according to the docs [here](https://nextjs.org/docs/advanced-features/i18n-routing#disabling-automatic-locale-detection). The reason why I want to disable it is because next-translate relies on it ([reference](https://github.com/vinissimus/next-translate/issues/578#issuecomment-817702029)).

**Comment by Jishnu-Prakash-K** [comment]
I think [18419](https://github.com/vercel/next.js/discussions/18419), might be the feature you are looking for.

**Comment by Jishnu-Prakash-K** [comment]
I think [18419](https://github.com/vercel/next.js/discussions/18419), might be the feature you are looking for.

**Comment by lalitindoria** [comment]
No that's different. I don't want any locales in my URL at all but I just want to use translations using next-translate.

**Comment by Yukioru** [comment]
Same problem. Can't turn off i18n routing if I don't need it.

**Comment by Nicolaschinc** [comment]
Similar problems, my code is: 
`i18n: {
    locales: ["en"],
    defaultLocale: "en",
    localeDetection: false,
  },`
When I type url like /en/xxxxx in browser address bar, and nextjs would automatically transform to /xxxx
How to fix it?

**Comment by zomars** [comment]
I'm bumping this issue as I'm currently facing the same issue and it is unexpected behavior.

**Comment by alexander-schreiner** [comment]
> No that's different. I don't want any locales in my URL at all but I just want to use translations using [next-translate](https://linear.app/vercel/issue/NEXT-translate).

Bumping after half a year of inactivity. I agree with the quoted comment: Imagine a dashboard which supports multiple locales. You probably don't want your URLs to be prefixed with the locale

**Comment by Kolby-Udacity** [comment]
I'm also facing this apparent bug.
